R/tm_missing_data.R

```diff
@@ -315,75 +316,61 @@ encoding_missing_data <- function(id, summary_per_patient = FALSE, ggtheme, data
       )
     ),
     hr(),
-    teal::get_rcode_ui(ns("rcode"))
+    teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code")
   )
 }
 
 #' @importFrom rlang .data
-srv_missing_data <- function(id, datasets, reporter, dataname, plot_height, plot_width, ggplot2_args) {
+srv_missing_data <- function(id, data, reporter, filter_panel_api, dataname, plot_height, plot_width, ggplot2_args) {
   with_reporter <- !missing(reporter) && inherits(reporter, "Reporter")
   moduleServer(id, function(input, output, session) {
-    teal.code::init_chunks()
-
     prev_group_by_var <- reactiveVal("")
 
-    raw_data <- reactive({
-      # unfiltered data maintains all attributes such as labels
-      datasets$get_data(dataname, filtered = FALSE)
-    })
+    data_r <- reactive({data[[dataname]]()})
 
-    data <- reactive({
-      datasets$get_data(dataname, filtered = TRUE)
-    })
-
-    data_keys <- reactive({
-      datasets$get_keys(dataname)
-    })
+    data_keys <- reactive(attr(data, "join_keys")$get(dataname)[[dataname]])
 
     # chunks needed by all three outputs stored here
-    common_code_chunks <- reactive({
-      # Create a private stack for this function only.
-      common_stack <- teal.code::chunks$new()
-      common_stack_push <- function(...) {
-        teal.code::chunks_push(..., chunks = common_stack)
-      }
-
+    common_code <- reactive({
       anl_name <- dataname
-      anl <- datasets$get_data(dataname, filtered = TRUE)
+      anl <- data[[dataname]]()
       assign(anl_name, anl)
-      teal.code::chunks_reset(chunks = common_stack)
       group_var <- input$group_by_var
 
-      if (!is.null(selected_vars()) && length(selected_vars()) != ncol(anl)) {
-        common_stack_push(
-          id = "ANL_selection_call",
-          expression = substitute(
+      nenv <- list()
+      nenv[["ANL"]] <- anl
+
+      quosure <- if (!is.null(selected_vars()) && length(selected_vars()) != ncol(anl)) {
+        teal.code::new_quosure(
```

Comment:
@Polkas 
this quosure has no data loading code. I think it's better to initialize the quosure with the new_quosure(data)

![image](https://user-images.githubusercontent.com/6959016/185400437-1e0886f0-b38d-40f9-a674-0251105d2868.png)

Reply:
It is needed as ANL is dynamic here.

Reply:
ANL is dynamic, but the whole preprocessing code is missing.